paying_for_college/disclosures/scripts/api_utils.py

```diff
@@ -11,38 +12,78 @@
 from decimal import Decimal
 
-from csvkit import CSVKitWriter as ckw
+import requests
+
+from paying_for_college.models import ConstantCap
+
+try:
+    LATEST_YEAR = ConstantCap.objects.get(slug='latest_year').value
+except:
+    LATEST_YEAR = 2013
 
+API_KEY = os.getenv('ED_API_KEY')
+API_ROOT = "https://api.data.gov/ed/collegescorecard/v1"
+SCHOOLS_ROOT = "%s/schools" % API_ROOT
+# SCHEMA_ROOT = "%s/data.json" % API_ROOT
+PAGE_MAX = 100  # the max page size allowed as of 2015-09-14
 
-def build_string(YEAR):
-    """assemble api fields for an api query string"""
-    fields = [
-        'id',
-        'ope8_id',
-        'ope6_id',
-        'school.name',
-        'school.city',
-        'school.state',
-        'school.zip',
-        'school.accreditor',
-        'school.school_url',
-        'school.degrees_awarded.predominant',
-        'school.degrees_awarded.highest',
-        'school.ownership',
-        'school.main_campus',
-        'school.branches',
-        'school.online_only',
-        'school.operating',
-        'school.under_investigation',
-        # detail fields delivered by year
+MODEL_MAP = {
+    'ope6_id': 'ope6_id',
+    'ope8_id': 'ope8_id',
+    '%s.student.size' % LATEST_YEAR: 'enrollment',
+    'school.accreditor': 'accreditor',
+    'school.school_url': 'url',
+    'school.degrees_awarded.predominant': 'degrees_predominant',  # data guide says this is INDICATORGROUP
+    'school.degrees_awarded.highest': 'degrees_highest',
+    'school.ownership': 'ownership',
+    'school.main_campus': 'main_campus',
+    # 'school.branches',  # ??
+    'school.online_only': 'online_only',
+    'school.operating': 'operating',
+    'school.under_investigation',
+}
+
+JSON_MAP ={
+    '%s.student.retention_rate.four_year.full_time' % LATEST_YEAR: 'RETENTRATE',
+    '%s.student.retention_rate.lt_four_year.full_time' % LATEST_YEAR: 'RETENTRATELT4',  # NEW
+    '%s.repayment.repayment_cohort.3_year_declining_balance' % LATEST_YEAR: 'REPAY3YR',  # NEW
+    '%s.repayment.3_yr_default_rate' % LATEST_YEAR: 'DEFAULTRATE',
+    '%s.aid.median_debt_suppressed.overall' % LATEST_YEAR: 'AVGSTULOANDEBT',
+    '%s.aid.median_debt_suppressed.completers.monthly_payments' % LATEST_YEAR: 'MEDIANDEBTCOMPLETER',  # NEW
+}
+
+BASE_STRING = [
+    'id',
+    'ope6_id',
+    'ope8_id',
+    'school.name',
+    'school.city',
+    'school.state',
+    'school.zip',
+    'school.accreditor',
+    'school.school_url',
+    'school.degrees_awarded.predominant',
+    'school.degrees_awarded.highest',
+    'school.ownership',
+    'school.main_campus',
+    'school.branches',
+    'school.online_only',
+    'school.operating',
+    'school.under_investigation',
+]
+
+
+def build_csv_fields(YEAR=LATEST_YEAR):
+    """assemble fields for an api query for an analysis csv"""
+    fields = BASE_STRING + [
         '%s.cost.attendance.academic_year' % YEAR,
         '%s.cost.attendance.program_year' % YEAR,
         '%s.cost.tuition.in_state' % YEAR,
```

Comment:
@higs4281 you may want to try using this since you are adding the same thing to each field, it may be less maintanence if you decided later to change YEAR to something else.
```python
fields = [
    'cost.attendance.academic_year',
    'cost.attendance.program_year',
    ....
]
fields = BASE_STRING  + ['%s.%s' % (YEAR, field) for field in fields]
```